bson/bsoncodec/default_value_decoders_test.go

```diff
@@ -2290,6 +2290,36 @@ func TestDefaultValueDecoders(t *testing.T) {
 				},
 			},
 		},
+		{
+			"CoreArrayDecodeValue",
+			defaultArrayCodec,
+			[]subtest{
+				{
+					"wrong type",
+					wrong,
+					nil,
+					&bsonrwtest.ValueReaderWriter{},
+					bsonrwtest.Nothing,
+					ValueDecoderError{
+						Name:     "CoreArrayDecodeValue",
+						Types:    []reflect.Type{tCoreArray},
+						Received: reflect.ValueOf(wrong),
+					},
+				},
+				{
+					"*bsoncore.Array is nil",
+					(*bsoncore.Array)(nil),
+					nil,
+					nil,
+					bsonrwtest.Nothing,
+					ValueDecoderError{
```

Comment:
Oh, perhaps I was wrong. Decoding `nil` does return an error then?

Reply:
What this is testing is a pointer to a `bsoncore.Array` that is `nil`. This is subtly different than a `bsoncore.Array` pointer to a `nil` value. If encoded, the latter would result in an empty BSON array. The former is not actually a settable value (a `nil` pointer cannot be referenced), and will therefore error on L39 of `array_codec`.